ldap/servers/slapd/plugin_syntax.c

```diff
@@ -500,16 +528,44 @@ slapi_mods_syntax_check(
                 /* Loop through the values and validate each one */
                 for (j = 0; mod->mod_bvalues[j] != NULL; j++) {
                     if (syntax_plugin->plg_syntax_validate(mod->mod_bvalues[j]) != 0) {
+                        char invalid_val[INVALID_BUFSIZ] = {0};
+                        char buffer[BUFSIZ];
+                        if (mod->mod_bvalues[j]->bv_len > 0) {
+                            if (mod->mod_bvalues[j]->bv_len >= INVALID_BUFSIZ) {
+                                if (mod->mod_bvalues[j]->bv_val[mod->mod_bvalues[j]->bv_len] != '\0') {
+                                    PR_snprintf(invalid_val, INVALID_BUFSIZ - 3,
+                                                "%s...", mod->mod_bvalues[j]->bv_val);
+                                } else {
+                                    /* strip last char/newline */
+                                    memcpy(&invalid_val,
+                                           mod->mod_bvalues[j]->bv_val,
+                                           INVALID_BUFSIZ-1);
+                                    /* Add truncation chars ... */
+                                    memcpy(&invalid_val[INVALID_BUFSIZ-4],
+                                           "...", 3);
+                                }
+                            } else {
+                                PR_snprintf(invalid_val, sizeof(invalid_val), "%s",
```

Comment:
Still a risk of read overflow about bval 
should probably:
  - use bv_len as size instead of sizeof(invalid_val)
  - invalid_val[bv_len] = '\0';